packages/nimble-components/src/utilities/models/tracker.ts

```diff
@@ -0,0 +1,62 @@
+type ObjectFromList<T extends readonly string[], V = string> = {
+    [K in T extends readonly (infer U)[] ? U : never]: V;
+};
+
+/**
+ * Generic Tracker which sets or resets flags and utilities
+ */
+export abstract class Tracker<WhimsList extends readonly string[]> {
```

Comment:
What's the benefit of having this class in addition to the `Validator` class? The only difference in the APIs is the extra `isValid` method on the `Validator`. Do we want the `Tracker` class for cases where the boolean state represents something other than validity?

Reply:
yes, the original tracker class was implemented for the update tracker in the table, we want to reuse it in the wafer map component. You can see more of the discussion in #1232. @rajsite suggested to look into reusing the same pattern for the column validator, and I tried to have the same pattern used in a generic validator class which can be also used in multiple components, not just the table columns